exercises/practice/bob/.approaches/answer-array/content.md

```diff
@@ -0,0 +1,40 @@
+# Answer array
+
+```csharp
+    // uses System.Linq;
+    private static readonly string[] answers = {"Whatever.", "Sure.", "Whoa, chill out!", "Calm down, I know what I'm doing!"};
+    
+    public static string Response(string statement)
+    {
+        var input = statement.TrimEnd();
+        if (input == "") return "Fine. Be that way!";
+        
+        var isShout = (input.Any(c => char.IsLetter(c)) && input.ToUpper() == input) ? 2: 0;
+        
+        var isQuestion = (input.EndsWith('?')) ? 1: 0;
+        
+        return answers[isShout + isQuestion];
+    }
+```
+
+An array can be defined that contains answers from which the right response is selected by an index calculated from scores given to the conditions.
```

Comment:
I really like that this approach has a _summary of the approach_ and then that it dives into details.

I think we should double down on that, and have this for all the single-topic approach documents.

We might even want to have a subheading to separate out the details from the summary.

In terms of the summary itself, this sentence might feel a bit overwhelming to someone who isn't super comfortable in English, or who are a bit new-ish to programming.

I'm thinking maybe there are some rules-of-thumb we could have on hand to help keep things a bit simpler.

My go-tos would be:
- Use 'you' instead of passive voice, if we can make it sound natural.
- Use shorter sentences, or sentences composed of shorter clauses.

So for the above I might say something like:

> In this approach you define an array that contains Bob’s answers, and each condition is given a score.
> The correct answer is selected from the array by using the score as the array index.

Reply:
@kytrinyx We used to have some docs (meaning we probably still do!) for automated feedback (and maybe mentoring too) that said we shouldn't use "you" because it could come across as accusatory. I think it's probably worth changing those guidelines now so that, at least in situations where we're not talking 1-1, we use more active voice and words like "you" because I agree that they parse better.

@ErikSchierboom Do you remember what I'm talking about?

Reply:
Ah, yes, for 1-1 automated feedback "you" can probably come across pretty strong.

When Sandi and I were working on the book we had a rule that we couldn't use "I" or "we", only "you".

For approaches there's an implicit understanding that you can take any of several approaches, so "you" feels less directive and more active, if that makes sense.

Reply:
@iHiD Yes I do remember. It's this document: https://exercism.org/docs/building/tooling/analyzers/comments#h-wording